src/index.coffee

```diff
@@ -0,0 +1,90 @@
+EvalSha = require 'redis-evalsha'
+_ = require 'underscore'
+async = require 'async'
+fs = require 'fs'
+redis = require 'redis'
+
+module.exports = class RateLimit
+
+  constructor: (@redisClient, rules, @prefix = 'ratelimit') ->
+    @checkFn = 'check_rate_limit'
+    @checkIncrFn = 'check_incr_rate_limit'
+
+    @eval = new EvalSha @redisClient
+    @eval.add @checkFn, @checkLimitScript()
+    @eval.add @checkIncrFn, @checkLimitIncrScript()
+
+    @rules = @convertRules rules
+
+  readLua: (filename) ->
+    fs.readFileSync("#{__dirname}/../lua/#{filename}.lua").toString()
+
+  checkLimitScript: ->
+    [
+      @readLua 'unpack_args'
+      @readLua 'check_limit'
+      'return 0'
+    ].join '\n'
+
+  checkLimitIncrScript: ->
+    [
+      @readLua 'unpack_args'
+      @readLua 'check_limit'
+      @readLua 'check_incr_limit'
+    ].join '\n'
+
+  convertRules: (rules) ->
+    for rule in rules
+      [rule.interval, rule.limit]
+
+  scriptArgs: (keys, weight = 1) ->
+    # Keys has to be a list
+    adjustedKeys = if _.isArray(keys) then keys else [keys]
+    adjustedKeys = _.chain(adjustedKeys)
+      .compact()
+      .filter(_.isString)
```

Comment:
I would combine this and the next filter as the extra iteration isn't saving you much readability here.